File: app/src/main/java/org/cnodejs/android/md/display/util/ToastUtils.java
package org.cnodejs.android.md.display.util;

import android.annotation.SuppressLint;
import android.content.Context;
import android.support.annotation.NonNull;
import android.support.annotation.StringRes;
import android.widget.Toast;

public final class ToastUtils {

    private volatile static ToastUtils singleton;

    public static ToastUtils with(@NonNull Context context) {
        if (singleton == null) {
            synchronized (ToastUtils.class) {
                if (singleton == null) {
                    singleton = new ToastUtils(context);
                }
            }
        }
        return singleton;
    }

    private final Toast toast;

    @SuppressLint("ShowToast")
    private ToastUtils(@NonNull Context context) {
        toast = Toast.makeText(context.getApplicationContext(), null, Toast.LENGTH_SHORT);
    }

    public void show(CharSequence msg) {
        toast.setText(msg);
        toast.show();
    }

    public void show(@StringRes int resId) {
        toast.setText(resId);
        toast.show();
    }

}

File: app/src/main/java/org/cnodejs/android/md/util/ToastUtils.kt
package org.cnodejs.android.md.util

import android.content.Context
import android.widget.Toast
import androidx.annotation.StringRes
import androidx.fragment.app.Fragment
import java.lang.ref.WeakReference

private var currentToast: WeakReference<Toast>? = null

fun Context.showToast(message: String) {
    currentToast?.get()?.cancel()
    val toast = Toast.makeText(applicationContext, message, Toast.LENGTH_SHORT).apply { show() }
    currentToast = WeakReference(toast)
}

fun Context.showToast(@StringRes resId: Int) {
    showToast(getString(resId))
}

fun Fragment.showToast(message: String) {
    context?.showToast(message)
}

fun Fragment.showToast(@StringRes resId: Int) {
    context?.showToast(resId)
}
